src/Microsoft.Identity.Client/TokenCache.cs

```diff
@@ -83,9 +83,15 @@ internal RequestContext CreateRequestContext()
         /// </summary>
         public TokenCache()
         {
+            var proxy = PlatformProxyFactory.CreatePlatformProxy(null);
             ServiceBundle = null;
-            TokenCacheAccessor = null;
-            LegacyCachePersistence = null;
+
+            // We have to create a "fake" telemetry manager here that won't have telemetry actually wired up so that 
+            // we can preserve the existing semantics of creating the TokenCache, performing operations on it such as Deserialize(), and then 
+            // attaching it to a ClientApplication.
+            // When we've fully moved to the builder-based API that doesn't take concrete TokenCache objects (and deprecated this public constructor), we can remove this.
+            TokenCacheAccessor = new TelemetryTokenCacheAccessor(new TelemetryManager(proxy, null, false), proxy.CreateTokenCacheAccessor());
```

Comment:
I'm confused about this being the telemetry token cache accessor.... is telemetry needed for this?